dmc2gym/wrappers.py

```diff
@@ -1,18 +1,21 @@
-from gym import core, spaces
-from dm_control import suite
-from dm_env import specs
+# MODIFIED BY Yawen Duan (https://github.com/kmdanielduan) to be able to
+# use `suite_module` to load environments from a dm_control based suite
+
 import numpy as np
+from dm_env import specs
+from gym import core, spaces
+from gym.utils import seeding
 
 
 def _spec_to_box(spec):
     def extract_min_max(s):
         assert s.dtype == np.float64 or s.dtype == np.float32
         dim = np.int(np.prod(s.shape))
         if type(s) == specs.Array:
-            bound = np.inf * np.ones(dim, dtype=np.float32)
+            bound = np.inf * np.ones(dim, dtype=np.float64)
```

Comment:
`float64` seems like more precision than is really needed for MuJoCo, curious what broke with `np.float32`? Also wondering if we should just have `dtype=s.dtype` instead?

Reply:
See summary of comments below.

Reply:
The original break was from a `seals` test for dtype match between `observation`(`float64`) and `observation_space`(`float32`). I have checked and found all observations and spaces `dtype` in dm_control are `np.float64`. Changing to `s.dtype` will results in converting to `np.float64`. 

I'm aware that [gym.mujoco_env](https://github.com/openai/gym/blob/150404104afc57ea618acca34d2d1f797d779869/gym/envs/mujoco/mujoco_env.py#L36) has its `observation_spaces.dtype` align with `observation.dtype` and then uses `np.float32` for the boundaries of the spaces.

Another point is that I noticed `_spec_to_box` may squash the dm_control observation `spec` (`OrderedDict`) into a flattened array. It's fine to do this to `action_spec`, but doing this to `observation_spec` will lose a lot more information. 

So, I plan to make further changes here to preserve the `Dict` structure of the observation when it's converted to space following [gym.mujoco_env](https://github.com/openai/gym/blob/150404104afc57ea618acca34d2d1f797d779869/gym/envs/mujoco/mujoco_env.py#L23).

Reply:
OK, gotcha, if `dm_control` uses float64 we should stick with that. If `s.dtype` works then it seems better to use that, so that if e.g. `dm_control` changes the default in future it'll propagate here?

Preserving `Dict` structure seems good, although I'll flag that I don't think any of `imitation` or most of SB3 is compatible with `Dict` observations currently. But we can either add that, or flatten it just before invoking the agent.